### PR TITLE
check KIDs for UIDs with cache

### PR DESCRIPTION
**Pull request by maxtaco**
- add a test
- we don't have a test for revoked keys yet, but we'd need a new test user

### CODE REVIEW

File: go/libkb/cached_user_loader.go [hunk 1]
```diff
@@ -83,6 +90,9 @@ func (u *CachedUserLoader) loadWithInfo(arg LoadUserArg, info *CachedUserLoadInf
 
 		var leaf *MerkleUserLeaf
 		leaf, err = lookupMerkleLeaf(u.G(), arg.UID, true, sigHints)
```
Comment (mlsteele) [inline go/libkb/cached_user_loader.go]: check `err` from `lookupMerkleLeaf` (same as comment in the other pr)

File: go/libkb/cached_user_loader.go [hunk 2]
```diff
@@ -114,3 +127,29 @@ func (u *CachedUserLoader) loadWithInfo(arg LoadUserArg, info *CachedUserLoadInf
...
 }
+
+func (u *CachedUserLoader) CheckKIDForUID(uid keybase1.UID, kid keybase1.KID) (found bool, revokedAt *keybase1.KeybaseTime, err error) {
+
+	var info CachedUserLoadInfo
+	upk, _, err := u.loadWithInfo(NewLoadUserByUIDArg(u.G(), uid), &info)
+	if err != nil {
+		return false, nil, err
+	}
+	if upk == nil {
```
Comment (mlsteele) [inline go/libkb/cached_user_loader.go]: Why could this happen? That `loadWithInfo() -> upk == nil && err == nil`?

File: go/libkb/upak.go [hunk 3]
```diff
@@ -18,3 +18,36 @@ func BaseProofSet(u *keybase1.UserPlusAllKeys) *ProofSet {
 	}
 	return NewProofSet(proofs)
 }
+
+// checkKIDPGP checks that the user has the given PGP KID valid *now*. Note that it doesn't
```
Comment (mlsteele) [inline go/libkb/upak.go]: I found the name `found` throughout this PR a little confusing. Here you say `valid`, I like that better as a name. But not a big deal, either is ok.

Comment (mlsteele) [inline go/libkb/upak.go]: ok

**Comment by maxtaco** [comment]
Thanks for the review @mlsteele! Ready for another look.

**Comment by keybase-ci-visdiff** [comment]
The commits 7ea5984c8afb...5c90130cedc8 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Pinentry-Paper_Key_Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479492193019-7ea5984c8afb-5c90130cedc8-linux/Pinentry-Paper_Key_Normal-7ea5984c8afb.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479492193019-7ea5984c8afb-5c90130cedc8-linux/Pinentry-Paper_Key_Normal-5c90130cedc8.png)  
   Pinentry-Paper_Key_Normal comparison
</summary>

**Comment by maxtaco** [comment]
`loadWithUser` never does after the above change.

`LoadUser` doesn't now, but might in the future, and don't want a crash then...

**Comment by maxtaco** [comment]
Actually, [it does](https://github.com/keybase/client/blob/35d843cc89bba0f2c066e4f71b02b4f5b2bd6776/go/libkb/loaduser.go#L190-L192)

**Comment by mlsteele** [comment]
Ok, thanks for looking

**Pull request merged**